### PR TITLE
summarazing point per approval institution

### PR DESCRIPTION
**Pull request by truhacevkir**
Cristin object looks like this:

Each **h_dbh_forskres_forfatter** has points for his affiliation. 
**CristinLocales** contains approvals for each topLevelOrganization. 

When crafting points, we summarize points for each **h_dbh_forskres_forfatter** at **institusjonsnr** and use topLevelOrg from CristinLocale to create institution points

```
{
  "publicationIdentifier": "018dc1025afe-4a3747e4-0f53-4602-84cb-10e66a941abd",
  "scientificResources": [
    {
      "h_dbh_forskres_forfatter": [
        {
          "personlopenr": "58612",
          "institusjonsnr": "186",
          "avdnr": "33",
          "undavdnr": "81",
          "gruppenr": "0"
        },
        {
          "personlopenr": "58613",
          "institusjonsnr": "186",
          "avdnr": "33",
          "undavdnr": "81",
          "gruppenr": "0"
        }
      ],
      "kvalitetsnivakode": "2",
      "arstall": "2011"
    }
  ],
  "cristinLocales": [
    {
      "eierkode": "UITO",
      "institusjonsnr": "186",
      "avdnr": "0",
      "undavdnr": "0",
      "gruppenr": "0",
      "brukernavn_kontrollert": "eer000",
      "dato_kontrollert": "2012-02-14",
      "status_kontrollert": "J"
    }
  ],
  "yearReported": "2011",
  "publicationDate": {
    "type": "PublicationDate",
    "year": "2011"
  },
  "instanceType": "AcademicChapter"
}
```

### CODE REVIEW

**Comment by truhacevkir** [comment]
There is made one assumption -> each contributor at affiliation has to have an approval for his institution, we will see if this assumption holds at next migration

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java [hunk 1]
```diff
@@ -48,22 +50,67 @@ public static DbCandidate toDbCandidate(CristinNviReport cristinNviReport) {
...
+    }
+
+    private static Collector<ScientificPerson, ?, Map<URI, BigDecimal>> collectToMapOfPoints(
+        List<CristinLocale> institutions) {
+        return Collectors.groupingBy(scientificPerson -> getTopLevelOrganization(scientificPerson, institutions),
+                                     Collectors.reducing(BigDecimal.ZERO, CristinMapper::toBigDecimal,
+                                                         BigDecimal::add));
+    }
+
+    private static BigDecimal toBigDecimal(ScientificPerson scientificPerson) {
```
Comment (monaullah) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/cristin/CristinMapper.java:103]: Rename method to `extractAuthorPointsForAffiliation`?

File: event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinMapperTest.java [hunk 2]
```diff
@@ -1,16 +1,98 @@
...
+        var scientificResource = scientificResourceWithCreators(creators);
+        var cristinLocale = cristinLocaleWithInstitutionIdentifier(institutionIdentifier);
+        var report = cristinReportFromCristinLocalesAndScientificResource(cristinLocale, scientificResource);
+        var dbCandidate = CristinMapper.toDbCandidate(report);
+
+        var institutionId = dbCandidate.points().get(0).institutionId();
+        var expectedInstitutionId = constructExpectedInstitutionId(cristinLocale);
+        var expectedPointsForInstitution = BigDecimal.ZERO
+                                               .add(new BigDecimal(POINTS_PER_CONTRIBUTOR))
+                                               .add(new BigDecimal(POINTS_PER_CONTRIBUTOR)).setScale(1);
```
Comment (monaullah) [inline event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinMapperTest.java:40]: In rest of nvi, the scale is 10 during calculations and 4 when persisting result of calculations. Maybe do the same here?

Comment (monaullah) [reply event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinMapperTest.java:40]: These values contain many decimals

**Review by monaullah — CHANGES_REQUESTED** [review]

File: event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinMapperTest.java [hunk 3]
```diff
@@ -37,7 +38,7 @@ void shouldUseCristinLocaleInstitutionWhenSummarizingPoints() {
         var expectedInstitutionId = constructExpectedInstitutionId(cristinLocale);
         var expectedPointsForInstitution = BigDecimal.ZERO
                                                .add(new BigDecimal(POINTS_PER_CONTRIBUTOR))
-                                               .add(new BigDecimal(POINTS_PER_CONTRIBUTOR)).setScale(1);
+                                               .add(new BigDecimal(POINTS_PER_CONTRIBUTOR)).setScale(4, RoundingMode.HALF_UP);
```
Comment (monaullah) [inline event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinMapperTest.java:41]: Suggestion for rewrite:
```
private static final BigDecimal POINTS_PER_CONTRIBUTOR = new BigDecimal("2.1398");
private static final int CALCULATION_PRECISION = 10;
private static final int SCALE = 4;

...
var expectedPointsForInstitution = POINTS_PER_CONTRIBUTOR
                                               .add(POINTS_PER_CONTRIBUTOR, new MathContext(CALCULATION_PRECISION, RoundingMode.HALF_UP))
                                               .setScale(SCALE, RoundingMode.HALF_UP);
...
```

**Pull request merged**